lib/components/src/TabsBar/components/ToolTab.tsx

```diff
@@ -0,0 +1,25 @@
+import { styled } from '@storybook/theming';
+import React, { FC, useContext, useEffect } from 'react';
+import { TabsBarContext } from '../TabsBarContext';
+import { TabsTool as _TabsTool } from '../TabsTool';
+import { TabProps } from '../types';
+import { getTabProps } from '../utils/get-tab-props';
+
+export const ToolTab: FC<TabProps> = (props) => {
+  const { addToMap, removeFromMap } = useContext(TabsBarContext);
+  const tabProps = getTabProps(props);
+
+  useEffect(() => {
+    addToMap(tabProps);
+
+    return () => {
+      removeFromMap(tabProps);
+    };
+  }, []);
```

Comment:
Can you explain this pattern to me?

It seems odd to me that this component adds itself to a context like this?

Reply:
The offset was the component `tabs` and it's usage - and especially `tabs` and the modified and borrowed and simulated usage across storybook.

This component (as it was) wanted to be able to just accept children and then pipe them into the whole logic train of the TabsBar, by not actually returning or using the given component/node, but creating a new, and altering props.

This pattern provides a hook-in to register to the context - via the context - and kind of yell "yes I want to be a part of this setup", and the TabsBar knows nothing of this and just accepts whatever wants to register with it.

The original usage is preserved and the syntax is for me more declarative and precise

This PR aims at TabsBar now being able to handle all of our tab bar requirements as we currently semi-mock and bypass many many places, as tabs with the older pattern can not handle what is required - where before it could only handle

- Tab with content

It can now handle

- Tab with content
- Tab as a button
- Tab as a button with a menu
- Tab that is just a separator
- And we can easily add to this range without having to do more than write that new tab component type.

It will then follow the pattern and syntax from `Accordion`

```ts
<TabsBar>
  <ContentTab id="canvas" label="Canvas">Show the canvas</ContentTab>
  <ContentTab id="docs" label="Docs">Show the docs</ContentTab>
  <SeparatorTab />
  <ToolTab id="zoom" icon="zoom" />
  <ToolTab id="zoom-out" icon="zoom-out" />
  <ToolTab id="zoom-reset" icon="zoom-reset" />
  <SeparatorTab />
  <ButtonTab id="grid" icon="grid" />
  <MenuTab id="viewport" icon="grow" menu={[{ id: 'small-mobile' label: 'Small Mobile'}]} />
</TabsBar>
```

Or you can do it as an config object

```ts
<TabsBar
  tabs={[
    { id: "canvas", label: "Canvas", content: "Show the canvas" },
    { id: "docs", label: "Docs", content: "Show the docs" },
    { type: "separator" },
    { id: 'zoom-in', type: 'tool', icon: 'zoom' },
    { id: 'zoom-out', type: 'tool', icon: 'zoom-out' },
    { id: 'zoom-reset', type: 'tool', icon: 'zoom-reset' },
    { type: "separator" },
    { id: 'grid', icon: 'grid', type: 'button' },
    {
      id: 'viewport',
      icon: 'grow',
      type: 'menu',
      menu: [
        { id: 'small-mobile', label: 'Small Mobile' },
        { id: 'mobile', label: 'Mobile' },
        { id: 'large-mobile', label: 'Large Mobile' },
        { id: 'tablet', label: 'Tablet' },
      ],
    },
  ]}
/>
```